apps/metadata-editor/src/app/edit/components/publish-button/publish-button.component.ts

```diff
@@ -42,9 +48,48 @@ export class PublishButtonComponent {
     })
   )
 
-  constructor(private facade: EditorFacade) {}
+  record$ = this.facade.record$
+  activeUser$: Observable<UserModel>
+
+  constructor(
+    private facade: EditorFacade,
+    private recordsApiService: RecordsApiService,
+    private platformService: PlatformServiceInterface,
+    private groupApiService: GroupsApiService
+  ) {
+    this.activeUser$ = this.platformService.getMe()
+  }
 
   saveRecord() {
     this.facade.saveRecord()
+    this.facade.saveSuccess$
+      .pipe(
+        take(1),
+        switchMap(() =>
+          combineLatest([this.activeUser$, this.record$]).pipe(take(1))
+        )
+      )
+      .subscribe(([userId, record]) => {
+        this.groupApiService.getGroups().subscribe((groups) => {
+          const groupId = groups.find(
+            (grp) => grp.name === record.ownerOrganization.name
+          )
+          // setting to 0 by default because the API expects a number and not null & most records don't have a group
+          this.recordsApiService
+            .setRecordOwnership(
+              record.uniqueIdentifier,
+              groupId ? groupId.id : 0,
+              Number(userId.id)
+            )
+            .subscribe(
```

Comment:
Sooo many subscribe... Can't we avoid this through better pipes?

Reply:
The groupApiService has been removed in my last commit. Can you tell me if the latest version is good with you or if it needs further rework? :)

Reply:
I would use a withLatestFrom to get the active user and record, as you only the current value for those, so no subscription needed there. And I would still try to put the set ownership api call inside the save pipe, with no added subscription.

Do you wanna try this? Or else I can take on.